File: lib/shared/models/frame_model.dart
class FrameModel {
  final String id;
  final String name;
  final String imageUrl;
  final int priceCoins;
  final int validityDays;
  final bool isVipOnly;

  FrameModel({required this.id, required this.name, required this.imageUrl, required this.priceCoins, required this.validityDays, required this.isVipOnly});

  factory FrameModel.fromJson(Map<String, dynamic> json) {
    return FrameModel(
      id: json['_id'] ?? '',
      name: json['name'] ?? 'Unknown Frame',
      imageUrl: json['imageUrl'] ?? '',
      priceCoins: json['priceCoins'] ?? 0,
      validityDays: json['validityDays'] ?? 30,
      isVipOnly: json['isVipOnly'] ?? false,
    );
  }
}

File: arvind-party-backend/src/modules/shop/frame.model.js
const mongoose = require('mongoose');

const frameSchema = new mongoose.Schema({
  name: { type: String, required: true },
  imageUrl: { type: String, required: true }, // Frame ki transparent PNG image
  priceCoins: { type: Number, required: true, default: 100 },
  validityDays: { type: Number, default: 30 }, // Kitne din tak chalega
  levelRequired: { type: Number, default: 1 }, // Optional: Level restriction
  isVipOnly: { type: Boolean, default: false },
  isActive: { type: Boolean, default: true }
}, { timestamps: true });

module.exports = mongoose.model('Frame', frameSchema);